bears/python/YapfBear.py

```diff
@@ -0,0 +1,137 @@
+import sys
+
+from coalib.bears.LocalBear import LocalBear
+from yapf.yapflib.yapf_api import FormatFile
+from coalib.results.Result import Result
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from coalib.results.Diff import Diff
+from coalib.misc.ContextManagers import prepare_file
+from coalib.bears.requirements.PipRequirement import PipRequirement
+
+
+class YapfBear(LocalBear):
+    """
+    Check and correct formatting of Python Code using ``yapf`` utility.
+
+    See <https://github.com/google/yapf> for more information.
+    """
+    LANGUAGES = {"Python"}
+    AUTHORS = {'The coala developers'}
+    REQUIREMENTS = {PipRequirement('yapf', '0.10')}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+
+    # TODO Add coalesce_brackets on supported by yapf
+    def run(self, filename, file,
+            max_line_length: int=79,
+            tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
+            allow_multiline_lambdas: bool=False,
+            blank_line_before_nested_class_or_def: bool=False,
+            continuation_tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
+            dedent_closing_brackets: bool=False,
+            indent_dictionary_value: bool=False,
+            join_multiple_lines: bool=True,
+            spaces_around_power_operator: bool=False,
```

Comment:
okay :)


2016-07-22 0:33 GMT+01:00 Zatreanu Adrian-Gabriel <notifications@github.com>
:

> In bears/python/YapfBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/578#discussion_r71806208>
> :
>
> > +    REQUIREMENTS = {PipRequirement('yapf', '0.10')}
> > +    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
> > +    LICENSE = 'AGPL-3.0'
> > +    CAN_FIX = {'Formatting'}
> > +
> > +    # TODO Add coalesce_brackets on supported by yapf
> > +    def run(self, filename, file,
> > +            max_line_length: int=79,
> > +            tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
> > +            allow_multiline_lambdas: bool=False,
> > +            blank_line_before_nested_class_or_def: bool=False,
> > +            continuation_tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
> > +            dedent_closing_brackets: bool=False,
> > +            indent_dictionary_value: bool=False,
> > +            join_multiple_lines: bool=True,
> > +            spaces_around_power_operator: bool=False,
>
> i think default should be True. i've seen
> 2 ** 3 used everywhere, whereas never
> 2**3
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/578/files/a5a49845b4cd44d17bf1e870c315e79fc82f753f#r71806208>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJ9xosHEFeFY5M4RcK_h0a4qLGxZDZ0jks5qYAHBgaJpZM4JMTJw>
> .
>